+nosnoc/+discrete_time_problem/Heaviside.m

```diff
@@ -91,6 +111,7 @@ function create_variables(obj)
             obj.w.z(0,0,opts.n_s) = {{'z', dims.n_z}, model.lbz, model.ubz, model.z0};
             obj.w.lambda_n(0,0,opts.n_s) = {{['lambda_n'], dims.n_lambda},0,inf};
             obj.w.lambda_p(0,0,opts.n_s) = {{['lambda_p'], dims.n_lambda},0,inf};
+            obj.w.alpha(0,0,opts.n_s) = {{['alpha'], dims.n_alpha},0, 1};
```

Comment:
I am wondering is this upper bound on alpha needed. It is also present in the complementarity constraints via 1-alpha, hence we have the constraint twice?

Reply:
We don't include $1-\alpha$ in the generic constraints $g$. so the upper box constraint is the unique $1-\alpha \ge 0$ constraint